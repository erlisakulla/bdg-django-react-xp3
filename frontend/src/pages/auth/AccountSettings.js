import React, { Component } from 'react';
import '../../css/Main.css';
import Navbar from '../components/Navbar.js';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Button, Card, Form, Nav } from 'react-bootstrap';

/**
 * Account Settings page
 *
 * @component
 */
class AccountSettings extends Component {
  render() {
    return (
      <> 
        <Navbar/>

        <div className="main-container">
          <h2>Account Settings</h2>
          <p>Here you can change your password or delete your account.</p>
          <hr/>
          
          <Card>
            <Card.Header>
              <Nav variant="tabs">
                <Nav.Item>
                  <Nav.Link href="/create">Setup Games</Nav.Link>
                </Nav.Item>
                <Nav.Item>
                  <Nav.Link href="/monitor">Monitor Games</Nav.Link>
                </Nav.Item>
                <Nav.Item>
                  <Nav.Link href="/join">Join Games</Nav.Link>
                </Nav.Item>
              </Nav>
            </Card.Header>

            <Card.Body>
              <div className="main-container-settings">
                <Form id="account_settings_form">
                  {/* ------- Current Password ------- */}
                  <Form.Group>
                    <Form.Control 
                      required
                      type="password"
                      id="current_password"
                      name="current_password"
                      placeholder="Current Password"/>
                  </Form.Group>
                  {/* ------- New Password ------- */}
                  <Form.Group>
                    <Form.Control 
                      required
                      type="password"
                      id="new_password"
                      name="new_password"
                      placeholder="New Password"/>
                  </Form.Group>
                  {/* ------- Repeat New Password ------- */}
                  <Form.Group>
                    <Form.Control 
                      required
                      type="password"
                      id="repeat_password"
                      name="repeat_password"
                      placeholder="Repeat New Password"/>
                  </Form.Group>
                  
                  <Form.Group>
                    <Button variant="primary" type="submit" id="setting_button1">
                      Change Password
                    </Button>
                  </Form.Group>
                </Form>
                <Button variant="danger" type="submit"  id="setting_button2">
                  Delete Account
                </Button>
              </div>
            </Card.Body>
          </Card>
        </div>
      </>
    );
  }
}

export default AccountSettings;