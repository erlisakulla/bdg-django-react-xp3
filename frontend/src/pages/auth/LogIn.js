import React, { Component } from "react";
import { Link } from "react-router-dom";
import 'bootstrap/dist/css/bootstrap.min.css';
import { Button, Form, Card, Nav } from 'react-bootstrap';
import axiosInstance from '../../axios'

/**
 * Login Form
 *
 * @component
 */
class LogIn extends Component {
  constructor(props) {
    super(props)

    this.onChangeInput = this.onChangeInput.bind(this);
    this.login = this.login.bind(this);

    /**
     * User data (/api/token)
     */
    this.state = {
      credentials: {
        email: '',
        password: '',
      }
    }
  }

  /**
   * Function for detecting input changes in the form
   *
   * @method
   * @param {Object} e event handler
   */
  onChangeInput = event => {
    const cred = this.state.credentials;
    cred[event.target.name] = event.target.value;
    this.setState({credentials: cred});
    // console.log(cred);
  }

  /**
   * Login function that handles posting the data
   *
   * @method
   * @param {Object} e event handler
   */
  login = (e) => {
    e.preventDefault();
    const loginUser = this.state.credentials;

    axiosInstance.post('http://127.0.0.1:8000/api/token/', loginUser, {crossDomain: true})
    .then((res) => {
      console.log(res);
      if (res.status === 200) {
        localStorage.setItem('access_token', res.data.access);
				localStorage.setItem('refresh_token', res.data.refresh);
				axiosInstance.defaults.headers['Authorization'] =
					'JWT ' + localStorage.getItem('access_token');
        
        window.location = "/create";
        alert("Successfully Logged In"); 
      } 
      else {
        console.log(res.data);
      }
    })
    .catch(error => {if(error.response){ console.log(error.response.data)}})
  }

  render() {
    return (
      <>
        <section className="content-wrapper">
          <div className="login">
            <h2 className="welcome-text">Welcome to the Beer Distribution Game!</h2>
            <div className="box">
              <Card id="forms">
                <Card.Header>
                  <Nav variant="tabs">
                    <Nav.Item>
                      <Nav.Link href="/" className="active">Login</Nav.Link>
                    </Nav.Item>
                    <Nav.Item>
                      <Nav.Link href="/signup">Register</Nav.Link>
                    </Nav.Item>
                  </Nav>
                </Card.Header>
                
                <Card.Body>
                  {/* Browser default validation using 'required' tag */}
                  <Form id="userCredentials" onSubmit={this.login}>
                    {/* ------- Email ------- */}
                    <Form.Group>
                      <Form.Control 
                        required
                        type="email"
                        id="email"
                        name="email"
                        value={this.state.credentials.email}
                        onChange={this.onChangeInput}
                        placeholder="Email"/>
                    </Form.Group>
                    {/* ------- Password ------- */}
                    <Form.Group>
                      <Form.Control 
                        required
                        type="password"
                        id="password"
                        name="password"
                        value={this.state.credentials.password}
                        onChange={this.onChangeInput}
                        placeholder="Password"/>
                    </Form.Group>

                    <Button variant="primary" type="submit" id="userSubmit">
                      Log In
                    </Button>
                    <Form.Text className="text-muted">
                      Don't have an account? <Link to="/signup">Register here.</Link>
                    </Form.Text>
                  </Form>
                </Card.Body>
              </Card>
            </div>
          </div>
        </section>
      </>
    );
  }
}

export default LogIn;