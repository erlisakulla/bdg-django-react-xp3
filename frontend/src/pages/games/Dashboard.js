import React from 'react';
import '../../css/Main.css';
import Navbar from '../components/Navbar';
import CreateGameForm from '../components/GameCreationForm';
import GamesList from '../components/GamesList';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Card, Nav } from 'react-bootstrap';
import axiosInstance from '../../axios'

/**
 * Dashboard page 
 *
 * @component
 */
class Dashboard extends React.Component {
  constructor(props) {
    super(props)

    this.componentDidMount = this.componentDidMount.bind(this);
    
    /**
     * Games display states
     */
    this.state = {
      is_instructor: '',
      games: null,
      message: 'No games to display',
    }
  }

  /**
   * Method to get all games and user data
   *
   * @method
   */
  componentDidMount() {
    axiosInstance.get('http://127.0.0.1:8000/api/game/')
    .then((res) => {
      const allGames = res.data;
      this.setState({games: allGames});
      console.log(res.data);
    })
    .catch(error => {if(error.response){console.log(error.response.data);}});

    axiosInstance.get('http://127.0.0.1:8000/api/user/')
    .then((res) => {
      const isInst = res.data.is_instructor;
      this.setState({is_instructor: isInst});
      console.log(res.data);
    })
    .catch(error => {if(error.response){console.log(error.response.data);}});
  }

  render() {
    return (
      <> 
        <Navbar/>
        
        <div className="main-container">
          <h2>Created Games</h2>
          <p>Here you can view, edit and delete all the games you have created.</p>
          <hr/>

          <Card>
            <Card.Header>
              <Nav variant="tabs">
                { this.state.is_instructor === true ?
                  <Nav.Item>
                    <Nav.Link href="/create" className="active">Setup Games</Nav.Link>
                  </Nav.Item> : null
                }
                { this.state.is_instructor === false ?
                  <Nav.Item>
                    <Nav.Link href="/join">Join Games</Nav.Link>
                  </Nav.Item> : null
                }
                <Nav.Item>
                  <Nav.Link href="/monitor">Monitor Games</Nav.Link>
                </Nav.Item>
              </Nav>
            </Card.Header>
            
            <Card.Body>
              { this.state.is_instructor === true ?
                <>
                  {/* Form for creating game */}
                  <CreateGameForm/>
                  {/* ------- Created Games Table ------- */}
                  <GamesList games={this.state.games} message={this.state.message}/>
                </> : 
                <h5 style={{textAlign:'center', fontWeight:'400', color:'grey'}}>Only instructors can create games</h5>
              }
            </Card.Body>
          </Card>
        </div>
      </>
    );
  }
}

export default Dashboard;