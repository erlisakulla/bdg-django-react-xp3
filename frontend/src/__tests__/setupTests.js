import '@testing-library/jest-dom';
import ReactDOM from 'react-dom';
import LogIn from '../pages/auth/LogIn';
import SignUp from '../pages/auth/Register';
import Dashboard from '../pages/games/Dashboard';
import MonitorGames from '../pages/games/MonitorGame';
import JoinGames from '../pages/games/JoinGame';
import AccountSettings from '../pages/auth/AccountSettings';
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

// Render and snapshot testing

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<LogIn/>, div);
});

it('matches snapshot', () => {
  const tree = renderer.create(<LogIn/>).toJSON();
  expect(tree).toMatchSnapshot();
});

////////////
  
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SignUp/>, div);
});

it('matches snapshot', () => {
  const tree = renderer.create(<SignUp/>).toJSON();
  expect(tree).toMatchSnapshot();
});

////////////

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Dashboard/>, div);
});

it('matches snapshot', () => {
  const tree = renderer.create(<Dashboard/>).toJSON();
  expect(tree).toMatchSnapshot();
});

////////////
 
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MonitorGames/>, div);
});

it('matches snapshot', () => {
  const tree = renderer.create(<MonitorGames/>).toJSON();
  expect(tree).toMatchSnapshot();
});

////////////
 
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<JoinGames/>, div);
});

it('matches snapshot', () => {
  const tree = renderer.create(<JoinGames/>).toJSON();
  expect(tree).toMatchSnapshot();
});

////////////
 
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AccountSettings/>, div);
});

it('matches snapshot', () => {
  const tree = renderer.create(<AccountSettings/>).toJSON();
  expect(tree).toMatchSnapshot();
});