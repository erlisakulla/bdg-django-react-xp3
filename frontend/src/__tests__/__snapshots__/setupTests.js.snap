// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="content-wrapper"
  data-testid="log-in-page"
>
  <div
    className="login"
  >
    <h2
      className="welcome-text"
    >
      Welcome to the Beer Distribution Game!
    </h2>
    <div
      className="box"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <img
        alt="Beer"
        className="beer-img"
        src="crate.png"
      />
      <div
        className="card"
        id="forms"
      >
        <div
          className="card-header"
        >
          <div
            className="nav card-header-tabs nav-tabs"
            onKeyDown={[Function]}
          >
            <div
              className="nav-item"
            >
              <a
                className="active nav-link"
                data-rb-event-key="/"
                href="/"
                onClick={[Function]}
                onKeyDown={[Function]}
              >
                Login
              </a>
            </div>
            <div
              className="nav-item"
            >
              <a
                className="nav-link"
                data-rb-event-key="/signup"
                href="/signup"
                onClick={[Function]}
                onKeyDown={[Function]}
              >
                Register
              </a>
            </div>
          </div>
        </div>
        <div
          className="card-body"
        >
          <form
            className=""
            id="userCredentials"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="email"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
                value=""
              />
            </div>
            <div
              style={
                Object {
                  "color": "red",
                  "paddingBottom": "10px",
                }
              }
            >
              
            </div>
            <button
              className="btn btn-primary"
              disabled={false}
              id="userSubmit"
              type="submit"
            >
              Log In
            </button>
            <small
              className="text-muted form-text"
            >
              Don't have an account? 
              <a
                href="/signup"
              >
                Register here.
              </a>
            </small>
          </form>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`matches snapshot 2`] = `
<section
  className="content-wrapper"
>
  <div
    className="login"
  >
    <h2
      className="welcome-text"
    >
      Welcome to the Beer Distribution Game!
    </h2>
    <div
      className="box"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <img
        alt="Beer"
        className="beer-img"
        src="crate.png"
      />
      <div
        className="card"
        id="forms"
      >
        <div
          className="card-header"
        >
          <div
            className="nav card-header-tabs nav-tabs"
            onKeyDown={[Function]}
          >
            <div
              className="nav-item"
            >
              <a
                className="nav-link"
                data-rb-event-key="/"
                href="/"
                onClick={[Function]}
                onKeyDown={[Function]}
              >
                Login
              </a>
            </div>
            <div
              className="nav-item"
            >
              <a
                className="active nav-link"
                data-rb-event-key="/signup"
                href="/signup"
                onClick={[Function]}
                onKeyDown={[Function]}
              >
                Register
              </a>
            </div>
          </div>
        </div>
        <div
          className="card-body"
        >
          <form
            className=""
            id="userCredentials"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="name"
                name="name"
                onChange={[Function]}
                placeholder="Full Name"
                required={true}
                type="text"
                value=""
              />
              <div
                className="invalid-feedback"
              >
                Invalid name.
              </div>
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                data-testid="email-input"
                id="email"
                name="email"
                onChange={[Function]}
                placeholder="Email"
                required={true}
                type="email"
                value=""
              />
              <div
                className="invalid-feedback"
              >
                Invalid email.
              </div>
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
                value=""
              />
              <div
                className="invalid-feedback"
              >
                Invalid password.
              </div>
            </div>
            <div
              className="form-group col"
            >
              <div
                className="form-row"
              >
                <div
                  className="form-group col"
                >
                  <div
                    className="radio"
                  >
                    <div
                      className="custom-control custom-radio"
                    >
                      <input
                        checked={false}
                        className="custom-control-input"
                        disabled={false}
                        id="is_instructor"
                        name="is_instructor"
                        onChange={[Function]}
                        required={true}
                        type="radio"
                        value="true"
                      />
                      <label
                        className="custom-control-label"
                        htmlFor="is_instructor"
                        title=""
                      >
                        Instructor
                      </label>
                    </div>
                  </div>
                </div>
                <div
                  className="form-group col"
                >
                  <div
                    className="radio"
                  >
                    <div
                      className="custom-control custom-radio"
                    >
                      <input
                        checked={false}
                        className="custom-control-input"
                        disabled={false}
                        id="is_student"
                        name="is_instructor"
                        onChange={[Function]}
                        type="radio"
                        value="false"
                      />
                      <label
                        className="custom-control-label"
                        htmlFor="is_student"
                        title=""
                      >
                        Student
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              style={
                Object {
                  "color": "red",
                  "paddingBottom": "10px",
                }
              }
            >
              
            </div>
            <button
              className="btn btn-primary"
              disabled={false}
              id="userSubmit"
              type="submit"
            >
              Sign Up
            </button>
            <small
              className="text-muted form-text"
            >
              Already have an account? 
              <a
                href="/"
              >
                Login here.
              </a>
            </small>
          </form>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`matches snapshot 3`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
    id="customNav"
  >
    <span
      className="navbar-brand"
      style={
        Object {
          "paddingRight": "10px",
        }
      }
    >
      Beer Distribution Game
    </span>
    <span
      className="navbar-text"
    >
      Logged in as 
      
    </span>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      />
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Logout
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/settings"
          href="/settings"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
            />
          </svg>
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="main-container"
  >
    <h2>
      Created Games
    </h2>
    <p>
      Here you can view, edit and delete all the games you have created.
    </p>
    <hr />
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        <div
          className="nav card-header-tabs nav-tabs"
          onKeyDown={[Function]}
        >
          <div
            className="nav-item"
          >
            <a
              className="nav-link"
              data-rb-event-key="/monitor"
              href="/monitor"
              onClick={[Function]}
              onKeyDown={[Function]}
            >
              Monitor Games
            </a>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      >
        <h5
          style={
            Object {
              "color": "grey",
              "fontWeight": "400",
              "textAlign": "center",
            }
          }
        >
          Only instructors can create games
        </h5>
      </div>
    </div>
  </div>,
]
`;

exports[`matches snapshot 4`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
    id="customNav"
  >
    <span
      className="navbar-brand"
      style={
        Object {
          "paddingRight": "10px",
        }
      }
    >
      Beer Distribution Game
    </span>
    <span
      className="navbar-text"
    >
      Logged in as 
      
    </span>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      />
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Logout
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/settings"
          href="/settings"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
            />
          </svg>
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="main-container"
  >
    <h2>
      Monitor Games
    </h2>
    <p>
      Here you can monitor all the games you have joined.
    </p>
    <hr />
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        <div
          className="nav card-header-tabs nav-tabs"
          onKeyDown={[Function]}
        >
          <div
            className="nav-item"
          >
            <a
              className="active nav-link"
              data-rb-event-key="/monitor"
              href="/monitor"
              onClick={[Function]}
              onKeyDown={[Function]}
            >
              Monitor Games
            </a>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      >
        <h5
          style={
            Object {
              "color": "grey",
              "fontWeight": "400",
              "textAlign": "center",
            }
          }
        >
          No games to display
        </h5>
      </div>
    </div>
  </div>,
]
`;

exports[`matches snapshot 5`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
    id="customNav"
  >
    <span
      className="navbar-brand"
      style={
        Object {
          "paddingRight": "10px",
        }
      }
    >
      Beer Distribution Game
    </span>
    <span
      className="navbar-text"
    >
      Logged in as 
      
    </span>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      />
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Logout
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/settings"
          href="/settings"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
            />
          </svg>
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="main-container"
  >
    <h2>
      Join Games
    </h2>
    <p>
      Here you can register to games, view and join all games you have registered for.
    </p>
    <hr />
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        <div
          className="nav card-header-tabs nav-tabs"
          onKeyDown={[Function]}
        >
          <div
            className="nav-item"
          >
            <a
              className="nav-link"
              data-rb-event-key="/monitor"
              href="/monitor"
              onClick={[Function]}
              onKeyDown={[Function]}
            >
              Monitor Games
            </a>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      >
        <h5
          style={
            Object {
              "color": "grey",
              "fontWeight": "400",
              "textAlign": "center",
            }
          }
        >
          Only students can join games
        </h5>
      </div>
    </div>
  </div>,
]
`;

exports[`matches snapshot 6`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
    id="customNav"
  >
    <span
      className="navbar-brand"
      style={
        Object {
          "paddingRight": "10px",
        }
      }
    >
      Beer Distribution Game
    </span>
    <span
      className="navbar-text"
    >
      Logged in as 
      
    </span>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      />
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rb-event-key={null}
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Logout
        </a>
        <a
          className="nav-link"
          data-rb-event-key="/settings"
          href="/settings"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            viewBox="0 0 24 24"
          >
            <path
              d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
            />
          </svg>
        </a>
      </div>
    </div>
  </nav>,
  <div
    className="main-container"
  >
    <h2>
      Account Settings
    </h2>
    <p>
      Here you can change your password or delete your account.
    </p>
    <hr />
    <div
      className="card"
    >
      <div
        className="card-header"
      >
        <div
          className="nav card-header-tabs nav-tabs"
          onKeyDown={[Function]}
        >
          <div
            className="nav-item"
          >
            <a
              className="nav-link"
              data-rb-event-key="/monitor"
              href="/monitor"
              onClick={[Function]}
              onKeyDown={[Function]}
            >
              Monitor Games
            </a>
          </div>
        </div>
      </div>
      <div
        className="card-body"
      >
        <div
          className="main-container-settings"
        >
          <form
            className=""
            id="account_settings_form"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="old_password"
                name="old_password"
                onChange={[Function]}
                placeholder="Current Password"
                required={true}
                type="password"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="New Password"
                required={true}
                type="password"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="password2"
                name="password2"
                onChange={[Function]}
                placeholder="Repeat New Password"
                required={true}
                type="password"
                value=""
              />
            </div>
            <small
              className="text-muted form-text"
              style={
                Object {
                  "paddingBottom": "15px",
                  "textAlign": "center",
                }
              }
            >
              Password must contain at least 8 characters.
            </small>
            <div
              className="form-group"
            >
              <button
                className="btn btn-primary"
                disabled={false}
                id="setting_button1"
                type="submit"
              >
                Change Password
              </button>
            </div>
          </form>
          <button
            className="btn btn-danger"
            disabled={false}
            id="setting_button2"
            type="submit"
          >
            Delete Account
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
